pkg/controller/atlasproject/cloud_provider_integration.go

```diff
@@ -224,51 +228,50 @@ func getCloudProviderIntegrations(projectSpec mdbv1.AtlasProjectSpec) []mdbv1.Cl
 	return projectSpec.CloudProviderIntegrations
 }
 
-func copyCloudProviderAccessData(cpiStatus *status.CloudProviderIntegration, atlasCPA *mongodbatlas.CloudProviderAccessRole) {
-	cpiStatus.AtlasAWSAccountArn = atlasCPA.AtlasAWSAccountARN
-	cpiStatus.AtlasAssumedRoleExternalID = atlasCPA.AtlasAssumedRoleExternalID
-	cpiStatus.RoleID = atlasCPA.RoleID
-	cpiStatus.CreatedDate = atlasCPA.CreatedDate
-	cpiStatus.AuthorizedDate = atlasCPA.AuthorizedDate
+func copyCloudProviderAccessData(cpiStatus *status.CloudProviderIntegration, atlasCPA admin.CloudProviderAccessAWSIAMRole) {
+	cpiStatus.AtlasAWSAccountArn = atlasCPA.GetAtlasAWSAccountArn()
+	cpiStatus.AtlasAssumedRoleExternalID = atlasCPA.GetAtlasAssumedRoleExternalId()
+	cpiStatus.RoleID = atlasCPA.GetRoleId()
+	cpiStatus.CreatedDate = timeutil.FormatISO8601(atlasCPA.GetCreatedDate())
+
+	if authorizedAt, ok := atlasCPA.GetAuthorizedDateOk(); ok {
+		cpiStatus.AuthorizedDate = timeutil.FormatISO8601(*authorizedAt)
+	}
 	cpiStatus.Status = status.CloudProviderIntegrationStatusCreated
 
-	if atlasCPA.AuthorizedDate != "" {
+	if _, ok := atlasCPA.GetAuthorizedDateOk(); ok {
 		cpiStatus.Status = status.CloudProviderIntegrationStatusAuthorized
 	}
 
-	if len(atlasCPA.FeatureUsages) > 0 {
-		cpiStatus.FeatureUsages = make([]status.FeatureUsage, 0, len(atlasCPA.FeatureUsages))
-
-		for _, feature := range atlasCPA.FeatureUsages {
-			if feature == nil {
-				continue
-			}
+	if len(atlasCPA.GetFeatureUsages()) > 0 {
+		cpiStatus.FeatureUsages = make([]status.FeatureUsage, 0, len(atlasCPA.GetFeatureUsages()))
 
+		for _, feature := range atlasCPA.GetFeatureUsages() {
 			id := ""
 
-			if feature.FeatureID != nil {
-				id = feature.FeatureID.(string)
+			if fID, ok := feature.GetFeatureIdOk(); ok {
+				id = fmt.Sprintf("%s.%s", fID.GetGroupId(), fID.GetBucketName())
```

Comment:
do we retain the format from before the migration?

Reply:
@s-urbaniak, No, but before making this change I verified what cause this struct to be changed and it's related to data lake(data federation), it seems that it has never been used by an operator user because it actually panics with the current implementation.

```
panic: interface conversion: interface {} is map[string]interface {}, not string [recovered]
        panic: interface conversion: interface {} is map[string]interface {}, not string
```

Reply:
i see, that's a great catch then and I see that exactly this piece of code is omitted by the unit tests:
<img width="813" alt="image" src="https://github.com/mongodb/mongodb-atlas-kubernetes/assets/375856/ee02dd13-bbfc-48f1-8a21-eaa5683a3a37">

Reply:
@helderjs in that case: please create a jira bug explaining the faulty behaviour, when it can occur and point it to this PR as well and describe under what cases a panic can occur.

please mark it as a blocker bug as well.

Reply:
Ticket created https://jira.mongodb.org/browse/CLOUDP-235207